service/frontend/workflowHandler.go

```diff
@@ -3011,37 +3018,594 @@ func (wh *WorkflowHandler) ListTaskQueuePartitions(ctx context.Context, request
+
+		if err := wh.validateTaskQueue(startWorkflow.TaskQueue); err != nil {
+			return nil, err
+		}
+
+		if err := wh.validateStartWorkflowTimeouts(&workflowservice.StartWorkflowExecutionRequest{
+			WorkflowExecutionTimeout: startWorkflow.WorkflowExecutionTimeout,
+			WorkflowRunTimeout:       startWorkflow.WorkflowRunTimeout,
+			WorkflowTaskTimeout:      startWorkflow.WorkflowTaskTimeout,
+		}); err != nil {
+			return nil, err
+		}
+
+		if len(startWorkflow.CronSchedule) > 0 {
+			return nil, errCronNotAllowed
+		}
+
+		if startWorkflow.WorkflowIdReusePolicy != enumspb.WORKFLOW_ID_REUSE_POLICY_UNSPECIFIED &&
+			startWorkflow.WorkflowIdReusePolicy != enumspb.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE {
+			return nil, errIDReusePolicyNotAllowed
+		}
+
+		// map search attributes to aliases here, since we don't go through the frontend when starting later
+		err = wh.processIncomingSearchAttributes(startWorkflow.GetSearchAttributes(), namespaceName)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// size limits will be validated on history. note that the start workflow request is
+	// embedded in the schedule, which is in the scheduler input. so if the scheduler itself
+	// doesn't exceed the limit, the started workflows should be safe as well.
+
+	// Set up input to scheduler workflow
+	input := &schedspb.StartScheduleArgs{
+		Schedule:     request.Schedule,
+		InitialPatch: request.InitialPatch,
+		State: &schedspb.InternalState{
+			Namespace:   namespaceName.String(),
+			NamespaceId: namespaceID.String(),
+			ScheduleId:  request.ScheduleId,
+		},
+	}
+	inputPayload, err := payloads.Encode(input)
+	if err != nil {
+		return nil, err
+	}
+	// Create StartWorkflowExecutionRequest
+	startReq := &workflowservice.StartWorkflowExecutionRequest{
+		Namespace:             request.Namespace,
+		WorkflowId:            request.ScheduleId,
+		WorkflowType:          &commonpb.WorkflowType{Name: scheduler.WorkflowType},
+		TaskQueue:             &taskqueuepb.TaskQueue{Name: scheduler.TaskQueueName},
+		Input:                 inputPayload,
+		Identity:              request.Identity,
+		RequestId:             request.RequestId,
+		WorkflowIdReusePolicy: enumspb.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE,
+		Memo:                  request.Memo,
+		SearchAttributes:      request.SearchAttributes,
+	}
+	_, err = wh.historyClient.StartWorkflowExecution(ctx, common.CreateHistoryStartWorkflowRequest(namespaceID.String(), startReq, nil, time.Now().UTC()))
+
+	if err != nil {
+		return nil, err
+	}
+	return &workflowservice.CreateScheduleResponse{
+		ConflictToken: scheduler.InitialConflictToken,
+	}, nil
 }
 
 // Returns the schedule description and current state of an existing schedule.
 func (wh *WorkflowHandler) DescribeSchedule(ctx context.Context, request *workflowservice.DescribeScheduleRequest) (_ *workflowservice.DescribeScheduleResponse, retError error) {
-	return nil, serviceerror.NewUnimplemented("todo")
+	defer log.CapturePanic(wh.logger, &retError)
```

Comment:
any idea what is typical latency for DescribeSchedule?

Reply:
in local testing it's been "fast enough". when it has to do a refresh it is slightly noticeable to me compared to the no-refresh case, so I'm guessing maybe 50-100ms more?